src/ui/public/courier/search_source/search_source.js

```diff
@@ -601,7 +603,16 @@ export function SearchSourceProvider(Promise, Private, config) {
             _.set(flatData.body, '_source.includes', remainingFields);
           }
 
-          flatData.body.query = buildESQuery(flatData.index, flatData.query, flatData.filters);
+          try {
+            flatData.body.query = buildESQuery(flatData.index, flatData.query, flatData.filters);
+          } catch (e) {
+            if (e.message === 'OutdatedKuerySyntaxError') {
+              const message =  `It looks like you're using an outdated Kuery syntax.
+                See what changed in the [docs](${documentationLinks.query.kueryQuerySyntax})!`;
+              throw new KbnError(message, KbnError);
```

Comment:
Rather than adding the message / link back here, would it be better to add a class for this error type to `errors.js`? It seems that's a convention we've followed elsewhere, but not sure if it's something we are sticking with.